runtime/src/accounts_file.rs

```diff
@@ -0,0 +1,179 @@
+use {
+    crate::{append_vec::*, storable_accounts::StorableAccounts},
+    solana_sdk::{account::ReadableAccount, clock::Slot, hash::Hash, pubkey::Pubkey},
+    std::{
+        borrow::Borrow,
+        io,
+        path::{Path, PathBuf},
+    },
+};
+
+#[derive(Debug)]
+/// An enum for accessing an accounts file which can be implemented
+/// under different formats.
+pub enum AccountsFile {
+    AppendVec(AppendVec),
+}
+
+impl AccountsFile {
+    pub fn new_append_vec(file: &Path, create: bool, size: usize) -> Self {
+        AccountsFile::AppendVec(AppendVec::new(file, create, size))
+    }
+
+    pub fn new_append_vec_from_file<P: AsRef<Path>>(
+        path: P,
+        current_len: usize,
+    ) -> io::Result<(Self, usize)> {
+        let (append_vec, len) = AppendVec::new_from_file(path, current_len)?;
+        Ok((AccountsFile::AppendVec(append_vec), len))
+    }
+
+    /// By default, all AccountsFile will remove its underlying file on
+    /// drop.  Calling this function to disable such behavior for this
+    /// instance.
+    pub fn set_no_remove_on_drop(&mut self) {
+        match self {
+            Self::AppendVec(av) => av.set_no_remove_on_drop(),
+        }
+    }
+
+    pub fn flush(&self) -> io::Result<()> {
+        match self {
+            Self::AppendVec(av) => av.flush(),
+        }
+    }
+
+    pub fn reset(&self) {
+        match self {
+            Self::AppendVec(av) => av.reset(),
+        }
+    }
+
+    pub fn remaining_bytes(&self) -> u64 {
+        match self {
+            Self::AppendVec(av) => av.remaining_bytes(),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        match self {
+            Self::AppendVec(av) => av.len(),
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        match self {
+            Self::AppendVec(av) => av.is_empty(),
+        }
+    }
+
+    pub fn capacity(&self) -> u64 {
+        match self {
+            Self::AppendVec(av) => av.capacity(),
+        }
+    }
+
+    pub fn file_name(slot: Slot, id: impl std::fmt::Display) -> String {
+        format!("{slot}.{id}")
+    }
+
+    /// Return (account metadata, next_index) pair for the account at the
+    /// specified `index` if any.  Otherwise return None.   Also return the
+    /// index of the next entry.
+    pub fn get_account(&self, index: usize) -> Option<(StoredAccountMeta<'_>, usize)> {
+        match self {
+            Self::AppendVec(av) => av.get_account(index),
+        }
+    }
```

Comment:
It might not be an "offset" for new formats.  I am trying to generalize here.

Reply:
If you're iterating one AccountsFile and want to get the ith account and the next one, this works for the new storage too.

Reply:
For now, I think we can just forward the API and land the PR to avoid addressing merge conflicts every time I rebase :joy:.  We will for sure change the API in the future esp. when we are revisiting the cache and in-memory representation of the account meta.